src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -81,45 +76,84 @@ class ExportEntitiesByTypeActor(rawlsDAO: RawlsDAO,
   def ExportEntities = streamEntities()
 
   /**
-    * Two basic code paths
-    *
-    * For Collection types, write the content to temp files, zip and return.
+    * Calls entitiesToTempFile() to write entity TSV to a temp file,
+    * then creates an HttpResponse containing that temp file's contents as an attachment.
     *
-    * For Singular types, pipe the content from `Source` -> `Flow` -> `Sink`
-    * Source generates the entity queries
-    * Flow executes the queries and sends formatted content to chunked response handler
-    * Sink finishes the execution pipeline
-    *
-    * Handle exceptions directly by completing the request.
+    * @see [[entitiesToTempFile()]]
     */
   def streamEntities(): Future[HttpResponse] = {
     val keepAlive = Connection("Keep-Alive")
 
+    entitiesToTempFile() map { tempFile =>
+      val (contentType, fileName) = if (modelSchema.isCollectionType(entityType)) {
+        (ContentTypes.`application/octet-stream`, s"$entityType.zip")
+      } else {
+        (ContentType(MediaTypes.`text/tab-separated-values`, HttpCharsets.`UTF-8`), s"$entityType.tsv")
+      }
+      val disposition = `Content-Disposition`.apply(ContentDispositionTypes.attachment, Map("filename" -> fileName))
+      HttpResponse(entity = HttpEntity.fromFile(contentType, tempFile.toJava), headers = List(keepAlive, disposition))
+    }
+  }.recoverWith {
+    // Standard exceptions have to be handled as a completed request
+    case t: Throwable => handleStandardException(t)
+  }
```

Comment:
I extracted the reusable part - saving TSV to a temp file - of the old `streamEntities()` method into a new reusable `entitiesToTempFile()` method. Then, `streamEntities()` and the new `streamEntitiesToWorkspaceBucket()` both invoke the reusable part.